src/app_engine/analytics_page.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+
+"""Module for the AnalyticsPage handler."""
+
+import json
+import time
+
+import webapp2
+
+import analytics
+import constants
+from constants import AnalyticsField
+
+
+class AnalyticsPage(webapp2.RequestHandler):
+  """Client Analytics data handler.
+
+  Each POST body to the AnalyticsPage is a JSON object of the form,
+  {
+    'request_time_ms': <client time in milliseconds when the request was sent>,
+    'type': <type of the content below>,
+    'content': <content related to the above type>
+  }
+
+  Currently the AnalyticsPage handles the following types:
+  'event': A JSON event object of the form:
+          {
+            'event_type': <string corresponding to an attribute of EventType>,
+            'event_time_ms': <client time when the event occurred>,
+            'room_id': <id of the room corresponding to the event [optional]>,
+          }
+
+      The event is reported to Analytics.report_event with the
```

Comment:
Can this be clarified? I read this several times and still didn't quite understand what offset you're calculating and for what purposes the timestamps are stored. When you say client time do you mean client local time? Or client UTC? Is the offset used to increase accuracy of reported time in some way to adjust for clock sync? Why do you need the original event time?